File: src/main/java/no/nav/melosys/eessi/models/sed/nav/Periode.java

package no.nav.melosys.eessi.models.sed.nav;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import lombok.Data;


@SuppressWarnings("unused")
@JsonInclude(Include.NON_NULL)
@Data
public class Periode {

    private AapenPeriode aapenperiode;

    private Fastperiode fastperiode;

    public boolean erAapenPeriode() {
        return aapenperiode != null && aapenperiode.getStartdato() != null;
    }
}

File: melosys-eessi-app/src/main/java/no/nav/melosys/eessi/models/sed/nav/Periode.kt
package no.nav.melosys.eessi.models.sed.nav

import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_NULL)
data class Periode(
    var aapenperiode: AapenPeriode? = null,
    var fastperiode: Fastperiode? = null
) {
    fun erAapenPeriode(): Boolean = aapenperiode != null && aapenperiode?.startdato != null
}
